src/import.rs

```diff
@@ -69,18 +69,18 @@ pub async fn import_many(
             let image = entries
                 .iter()
                 .enumerate()
-                .find(|(_, entry)| {
-                    entry
-                        .entry()
-                        .filename()
-                        .as_str()
-                        .map(|name| name == "icon.png" || name == "icon.jpg")
-                        .unwrap_or_default()
+                .map(|(index, entry)| {
+                    (index, entry.entry().filename().as_str().unwrap_or_default())
                 })
-                .map(|a| a.0);
-            let image = if let Some(index) = image {
+                .find(|(_, name)| *name == "icon.png" || *name == "icon.jpg");
+            let image = if let Some((index, name)) = image {
                 let res = read_vec(&reader, index).await?;
-                encode(&res)
+                let ending = name
+                    .split('.')
+                    .nth(1)
+                    .context(format!("Can't extract file ending from {name}"))?;
+                let base64 = encode(&res);
+                format!("data:image/{ending};base64,{base64}")
```

Comment:
this is wrong. the mimetype for .jpg files is `image/jpeg`, see eg. https://www.w3.org/Graphics/JPEG/

although browser will probably correct the fault, i'd go for "be strict in what you write, be liberal in what you accept from others"